langchain/chains/base.py

```diff
@@ -1,13 +1,38 @@
 """Base interface that all chains should implement."""
 from abc import ABC, abstractmethod
-from typing import Any, Dict, List
+from typing import Any, Dict, List, Union
 
-from pydantic import BaseModel
+from pydantic import BaseModel, Extra
+
+
+class Memory(BaseModel, ABC):
+    """Base interface for memory in chains."""
+
+    class Config:
```

Comment:
oh woah can you just put this on base class and have it apply to all subclasses? i need to do that!